### PR TITLE
Nicer error message is symlinking chokes on an existing file 

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/shared/install.c [hunk 1]
```diff
@@ -382,8 +388,10 @@ static int create_symlink(
         }
 
         r = symlink_atomic(old_path, new_path);
-        if (r < 0)
-                return r;
+        if (r < 0) {
+                unit_file_changes_add(changes, n_changes, r, new_path, NULL);
+                return -EEXIST;
```
Comment (poettering) [inline src/shared/install.c]: THis converts all errors into EEXIST here, is that intended?

**Comment by poettering** [comment]
Hmm, the second patch of yours removes stuff that was there for the SELinux support. Which I removed a while back, because it was incorrect... (because it was sitting before the install calls and searched for unit files on its own to read a label from them) The idea was always to readd correct SELinux support later on, see #1050... But I figure it's OK removing this for now in full, I guess should the correct SELinux code come back later it probably won't need the verb passed in like this anyway...

**Comment by poettering** [comment]
Wondering about the EEXIST thin, but looks ready to merge in general.

File: src/shared/install.c (hunk 1 above)
Comment (keszybz) [inline src/shared/install.c]: Nope. Will fix.

**Comment by keszybz** [comment]
Whoever works on #1050 should then probably start by reverting this patch... But I don't see the point of keeping some half-broken code, so I think it's fine to remove it for now.

**Comment by keszybz** [comment]
Repushed.

**Comment by keszybz** [comment]
Two more patches... I think that's the right thing to do, but not 100% certain.

**Pull request merged**